src/service/server/QueryHandler.cpp

```diff
@@ -240,30 +244,40 @@ std::shared_ptr<Query> QueryHandler::buildQuery(const Table& tb, const QueryRequ
 
     // we have minimum size of window as 1 second to be enforced
     // so if buckets is smaller than range (seconds), we use each range as
-    auto range = req.end() - req.start();
+    auto range = (req.end() - req.start());
+    LOG(INFO) << "range is: " << range;
     N_ENSURE_GT(range, 0, "timeline requires end time greater than start time");
 
-    int32_t window = (int32_t)req.window();
+    int64_t window = (int64_t)req.window();
+    int64_t time_unit = (int64_t)req.time_unit();
+  
     auto buckets = window == 0 ? FLAGS_AUTO_WINDOW_SIZE : range / window;
+    LOG(INFO) << FLAGS_AUTO_WINDOW_SIZE;
+    LOG(INFO) << "number of buckets: " << buckets;
+    // LOG(INFO) << "window: " << window;
+    LOG(INFO) << "time unit: " << time_unit;
     if (buckets == 0 || buckets > range) {
       buckets = range;
     }
-
     // recalculate window based on buckets
     window = range / buckets;
     N_ENSURE_GT(window, 0, "window should be at least 1 second");
-
+    // LOG(INFO) << "recalculated window: " << window; 
     // only one bucket?
     std::shared_ptr<Expression> windowExpr = nullptr;
-    if (buckets < 2) {
```

Comment:
clean up and revert all the formatting changes will help review without the noisy distractions.